extensions/packageManager/src/node/feeds/nuget/nugetFeed.ts

```diff
@@ -0,0 +1,127 @@
+import { IFeed, IPackageQuery, IPackageDefinition, IPackageSource } from '../feedInterfaces';
+import { INuGetSearchResult, INuGetServiceIndex } from './nugetInterfaces';
+import axios from 'axios';
+import * as semverSort from 'semver-sort';
+
+/**
+ * NuGet v3 feed.
+ */
+export class NuGetFeed implements IFeed {
+  private language: string = 'c#';
+  private source: string = 'nuget';
+
+  /**
+   * Creates a NuGet v3 feed
+   * @param composer Composer registration for logging.
+   * @param packageSource Package source that represents the target NuGet source.
+   * @param data Optional, legacy data parameter for cases in which we already have the nuget result.
+   */
+  constructor(
+    private composer: IExtensionRegistration,
+    private packageSource?: IPackageSource,
+    private data?: INuGetSearchResult | INuGetServiceIndex
+  ) {}
+
+  /**
+   * Gets packages from the NuGet feed using the package source default query, or if provided, the query parameter.
+   * @param query Optional query parameter to be used when searching the NuGet feed. If not provied, we use the default source query.
+   */
+  async getPackages(query?: IPackageQuery): Promise<IPackageDefinition[]> {
+    // Legacy: If seed data was provided on construction, use that instead of calling the api
+    if (this.data) {
+      // Is the seed data a search result? Note that ideally we want to remove this codepath since we cannot
+      // do paging or other more advanced operations. Since this path is very inflexible, also
+      // ignores the query provided in the package query.
+      // Example url that ends up in this path: https://azuresearch-usnc.nuget.org/query?q=Tags:%22bf-component%22&prerelease=true
+      const searchResult = this.data as INuGetSearchResult;
+      if (searchResult.data) {
+        return this.asPackageDefinition(searchResult);
+      }
+    }
+
+    // Query package source url
+    // Example url that ends in this path: https://www.myget.org/F/ms-test-adapters/api/v3/index.json
+    let serviceIndex: INuGetServiceIndex = this.data as INuGetServiceIndex;
+
+    // If we don't have a service index provided at construction, get it from the url
+    if (!serviceIndex || !serviceIndex.resources) {
+      const httpResponse = await axios.get(this.packageSource.url);
+      serviceIndex = httpResponse?.data;
+    }
+
+    // Now that we have a service index either from seed data or from querying the NuGet feed,
+    // we get the query endpoint, build query parameters, call feed api and return query results to caller
+    const queryEndpoint = serviceIndex?.resources?.find((resource) => resource['@type'] === 'SearchQueryService');
+    if (queryEndpoint) {
+      const nuGetSearchUrl = this.buildNuGetSearchUrl(queryEndpoint['@id'], query);
+
+      this.composer.log(`Retrieving NuGet packages with computed search url ${nuGetSearchUrl}`);
+      const httpResponse = await axios.get(nuGetSearchUrl);
+
+      if (!httpResponse || !httpResponse.data) {
```

Comment:
`if (httpResponse?.data != null)` would work here (or `if (httpResponse?.data)` if there's no way this can be tricked by a falsy non-null value).

Reply:
So here I want the opposite logical result (enter if something is null) but I get your point of doing a single conditional statement.